### PR TITLE
TIL how to yank words to registers in VIM

### PR DESCRIPTION
**Pull request by jonathanyeong**
I had issues with my deleted words always replacing my yanked word. Today I finally found out that you can save a yanked word to a certain register. This is just a basic way to ensure that your yanked text isn't overwritten by other yanks or deletes.

### CODE REVIEW

**Comment by jonathanyeong** [comment]
I hope this is the type of TIL you're looking for! Also very open to better ways of doing this.

File: vim/yanking-text-to-register.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Yanking text to register
+
+Yanking and deleting text will save the text to the same unamed register. Which means every time you delete a word, your 
+yanked word will be overwritten. Which caused me endless amounts of frustration when I accidentally deleted a word and had to
+remember what I yanked earlier.
+
+I learned that you can specify a register to save your yanked word to by prepending your yank command with `"a`. Where a
```
Comment (JoelQ) [inline vim/yanking-text-to-register.md]: Thoughts on wrapping the `a` in backticks so it stands out? "Where a is a register" should probably be part of the previous sentence.

> I learned that you can specify a register to save your yanked word to by prepending your yank command with `"a`, where `a` is a register. There are ...

File: vim/yanking-text-to-register.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Yanking text to register
+
+Yanking and deleting text will save the text to the same unamed register. Which means every time you delete a word, your
```
Comment (srt32) [inline vim/yanking-text-to-register.md]: I think `unamed` should be replaced with `unnamed`

File: vim/yanking-text-to-register.md (hunk 1 above)
Comment (jonathanyeong) [inline vim/yanking-text-to-register.md]: Fixed in commit 132431540ba9fe40a9aa5eaaf5135a2ba0ce6081.

File: vim/yanking-text-to-register.md (hunk 2 above)
Comment (jonathanyeong) [inline vim/yanking-text-to-register.md]: Fixed in commit 132431540ba9fe40a9aa5eaaf5135a2ba0ce6081

**Comment by zamith** [comment]
@jonathanyeong You can also "save" a deletion to the `_`, which is the `/dev/null` of registers. Here's an example:

```
yw        # Save a word to the unnamed register
"_dw    # Delete a word and "save" to the _ register
```

If you then try to `"_p`, nothing will happen, because the `_` register never saves anything.

This is also very useful when your yanking occurs outside of vim (comes from the clipboard), which you can have by setting your unnamed register to the clipboard, like this:

```
set clipboard=unnamed
```

File: vim/yanking-text-to-register.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Yanking text to register
+
+Yanking and deleting text will save the text to the same unnamed register. Which means every time you delete a word, your
```
Comment (JoelQ) [inline vim/yanking-text-to-register.md]: How about replacing "Which means.." with  "This means..." or even just "Every time you delete..." ?

File: vim/yanking-text-to-register.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Yanking text to register
+
+Yanking and deleting text will save the text to the same unnamed register. Which means every time you delete a word, your 
+yanked word will be overwritten. Which caused me endless amounts of frustration when I accidentally deleted a word and had to
```
Comment (JoelQ) [inline vim/yanking-text-to-register.md]: "This caused me..." ?

File: vim/yanking-text-to-register.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+I learned that you can specify a register to save your yanked word to by prepending your yank command with `"a`. Where `a`
+is a register. There are a number of different registers which do different things. If you want to learn more read vim's [register
+documentation](http://vimdoc.sourceforge.net/htmldoc/change.html#registers).
+
+An example usage is:
+
+`"ayw`
+
+This will save the yanked word to the register a. After deleting various things in your file you can always paste your yanked
```
Comment (JoelQ) [inline vim/yanking-text-to-register.md]: Wrap `a` in backticks

Comment (jonathanyeong) [inline vim/yanking-text-to-register.md]: Ah missed that one. Thanks for picking that up. Fixed in commit 4978896c50842e90dd6e546989ce91da16201049.

File: vim/yanking-text-to-register.md (hunk 4 above)
Comment (jonathanyeong) [inline vim/yanking-text-to-register.md]: Yup 'this' makes more sense. Fixed in commit 1a3888ced8c199079df6ee67da5de4fb9dc50cef.

File: vim/yanking-text-to-register.md (hunk 3 above)
Comment (jonathanyeong) [inline vim/yanking-text-to-register.md]: I just chose "Every time you delete...". Fixed in commit 4978896c50842e90dd6e546989ce91da16201049. Thanks!

**Comment by jonathanyeong** [comment]
Great tip @zamith! That's another good way to stop the deleted word from overwriting your yanked word. Also, I had no idea you could do that with the clipboard! Do you mind if I add it into the TIL?

**Comment by iwz** [comment]
This feature is vim is enhanced by the `:registers` command which shows you what's in each register.

**Comment by JoelQ** [comment]
:+1:  for `:registers`

**Comment by zamith** [comment]
@jonathanyeong Go ahead. ;)

**Comment by jonathanyeong** [comment]
Thanks @zamith and @iwz. I've added your tips to the file! The `:registers` command is a great command to know about!

**Comment by ferrouswheel** [comment]
If you don't want to remap your clipboard register, you can usually interact with the system clipboard by using the `*` register, e.g. to paste the clipboard contents: `"*p`

**Comment by JoelQ** [comment]
I've merged this as https://github.com/thoughtbot/til/commit/e7356b58e1c5def566e221b5470e18e1348c60d0

We try to keep tips short and focused so I removed the section on the null register suggested by @zamith . I didn't want this to turn into a tutorial on VIM's registers.

Thanks for contributing!

**Pull request closed without merging**